docs/inheritance_and_composition.md

```diff
@@ -0,0 +1,36 @@
+# Template Inheritance and Composition
+
+## What is Inheritance, what is Composition?
+
+Template Inheritance is a way to create a new template based on an existing one. It
```

Comment:
* I'd not capitalize "Inheritance" in a regular sentence.
* How about introducing "parent" and "child" in the first sentence to clarify the relationship for later use?

```suggestion
Template inheritance is a way to create a new (child) template based on an existing (parent) template. It
```